Guide/htmx-and-hyperscript.markdown

```diff
@@ -0,0 +1,178 @@
+# htmx and hyperscript
+
+```toc
+
+```
+
+## Introduction
+
+As IHP itself is a hypermedia based tool, defaulting on hyperlinks, forms and RESTful routes, it works well with [htmx](https://htmx.org/) and [hyperscript](https://hyperscript.org/).
+
+These tools can in many cases entirely replace the need for adding Single Page Applications (SPA) and npm to your workflow.
+
+Instead of the separation of concerns paradigm, htmx and hyperscript favours [Locality of Behaviour](https://htmx.org/essays/locality-of-behaviour/).
+
+To learn more, htmx has an [excellent collection of essays](https://htmx.org/essays/) about using the hypermedia approach instead of the current SPA paradigm.
+
+## Make it play nice Turbolinks and AutoRefresh
+
+To have htmx and hyperscript play well with Turbolinks page transitions, you can add this to your main javascript file.
+
+```javascript
+document.addEventListener('turbolinks:load', () => {
+    htmx.process(document.body);
+    _hyperscript.processNode(document.body);
+});
+```
+
+This makes sure that htmx and hyperscript code does not stop working after a Turbolinks page transition or an AutoRefresh.
+
+## htmx
+
+htmx gives you access to AJAX, as a way to update views locally with an API conviently residing in HTML attributes.
+
+Instead of the typical Single Page Application pattern parsing JSON and turning into html in the DOM, the API endpoints simply just return HTML to be patched into the DOM.
+
+### Installation
+
+The recommended way of installing htmx is downloading [htmx.min.js](https://unpkg.com/htmx.org/dist/htmx.min.js) from unpkg.com and save it to `static/vendor/htmx.min.js` in your project directory.
```

Comment:
For Tailwind we recommend [using npm](https://github.com/digitallyinduced/ihp/blob/efeb31cd35c6f62effcaa82dac4aa45e2a322e02/Guide/tailwindcss.markdown#installing-tailwind) - can we do something similar here?

Reply:
Yes, it's very doable with the standalone cli which also has all official plugins like the form plugin already built in: https://tailwindcss.com/blog/standalone-cli

Only problem with that is cross-platform support as one binary for each system would be required.

Would be great if some NixOS maintainers would add it in their package registry (they only have the nodePackages one without integrated plugins) as I don't have the energy for it right now :) But shouldn't be too complex I think